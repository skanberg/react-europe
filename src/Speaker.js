import React from "react";
import { Row, Col, Block } from "jsxstyle";

const Speaker = ({ avatarUrl, name, twitter }) => (
  <Row marginBottom={10}>
    <img
      style={{ width: 50, height: 50, borderRadius: "50%", marginRight: 10 }}
      src={avatarUrl}
      alt=""
    />
    <Col justifyContent="space-around">
      <Block fontWeight={400}>{name}</Block>
      <Block fontWeight={300}>@{twitter}</Block>
    </Col>
  </Row>
);

export default Speaker;
