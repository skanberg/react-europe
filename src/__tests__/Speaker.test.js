import React from "react";
import renderer from 'react-test-renderer';
import Speaker from "../Speaker";

test("Speaker renders correctly", () => {
  const tree = renderer.create(
    <Speaker
      avatarUrl="http://server/image.png"
      name="name"
      twitter="twitter"
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
