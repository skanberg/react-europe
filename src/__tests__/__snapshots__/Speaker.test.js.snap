exports[`test Speaker renders correctly 1`] = `
<div
  className="d-flex flex-direction-row mb-10 _img-w-50 _img-h-50 _img-border-radius-50P _img-mr-10">
  <img
    alt=""
    src="http://server/image.png" />
  <div
    className="d-flex flex-direction-column _1bdo1uw">
    <div
      className="fw-400">
      name
    </div>
    <div
      className="fw-300">
      @
      twitter
    </div>
  </div>
</div>
`;
